### PR TITLE
Move `experimental/oak_baremetal_kernel` to `restricted_kernel`.

### PR DESCRIPTION
**Pull request by andrisaar**
We now have a name for it, and we know we're going to use it long-term, therefore experimental has been the wrong home for it for quite some time now.

### CODE REVIEW

File: experimental/oak_baremetal_app_crosvm/Cargo.toml [hunk 1]
```diff
@@ -11,13 +11,13 @@ members = ["."]
...
-vsock_channel = ["oak_baremetal_kernel/vsock_channel"]
-simple_io_channel = ["oak_baremetal_kernel/simple_io_channel"]
+vsock_channel = ["oak_restricted_kernel/vsock_channel"]
+simple_io_channel = ["oak_restricted_kernel/simple_io_channel"]
 
 [dependencies]
 log = "*"
 oak_baremetal_communication_channel = { path = "../oak_baremetal_channel" }
-oak_baremetal_kernel = { path = "../../experimental/oak_baremetal_kernel", default-features = false }
+oak_restricted_kernel = { path = "../../restricted_kernel", default-features = false }
```
Comment (conradgrobler) [inline experimental/oak_baremetal_app_crosvm/Cargo.toml:20]: For consistency the crate name and folder name should probably the same (similar to #3275).

Comment (tiziano88) [reply experimental/oak_baremetal_app_crosvm/Cargo.toml:20]: Yes please

Comment (andrisaar) [reply experimental/oak_baremetal_app_crosvm/Cargo.toml:20]: I still think the `oak_` prefix is visual noise at the top level, but I've been outvoted :)

**Review by mariaschett — APPROVED** [review]
Great, thanks!

**Pull request merged**